nemo_curator/modules/semantic_dedup/clusteringmodel.py

```diff
@@ -128,7 +128,22 @@ def __call__(self, embeddings_dataset: DocumentDataset):
             embeddings_df = embeddings_df.repartition(
                 partition_size=self.partition_size
             )
-            embeddings_df = embeddings_df.to_backend("pandas").persist()
+
+            try:
+                embeddings_df = embeddings_df.to_backend("pandas").persist()
+
+                if embeddings_df.shape[0].compute() < self.n_clusters:
+                    raise ValueError(
+                        "Number of clusters is greater than the number of documents in your dataset. "
+                        "Please reduce n_clusters to be less than or equal."
+                    )
```

Comment:
@VibhuJawa would be interested in your opinion here. Do you think checking npartitions, the length of a single partition, etc. would be a sufficient and less computationally expensive way to check this?

Reply:
I think this works (because above is persisted all ready), we are not really causing a extra computation apart from a size calculation. 

We should probably improve the error message (explictly telling the number of documents, number of clusters set) etc, because it will help the user fix it based on the message directly than digging inside the code.

Reply:
Good point, updating.